src/System.Private.Xml/src/System/Xml/BinHexDecoder.cs

```diff
@@ -164,7 +175,11 @@ public static unsafe byte[] Decode(char[] chars, bool allowOddChars)
 
             if (hasHalfByteCached && !allowOddChars)
             {
+#if XMLSERIALIZERGENERATOR
+                throw new XmlException(SR.Format(SR.Xml_InvalidBinHexValueOddCount, new string(chars)));
+#else
                 throw new XmlException(SR.Xml_InvalidBinHexValueOddCount, new string(chars));
+#endif
```

Comment:
I have to make a copy of strings.resx since the current build system always auto add the strings.resx under the resource folder of the project to the project. That means you are not allowed to refer a strings.resx in your project file.